sdk/servicebus/Azure.Messaging.ServiceBus/src/Core/BasicRetryPolicy.cs

```diff
@@ -69,6 +69,14 @@ public BasicRetryPolicy(ServiceBusRetryOptions retryOptions)
         ///
         public override TimeSpan CalculateTryTimeout(int attemptCount) => Options.TryTimeout;
 
+        /// <summary>
+        ///   Gets the maximum number of retry attempts configured for this policy.
+        /// </summary>
+        ///
+        /// <returns>The maximum number of retry attempts.</returns>
+        ///
+        protected override int GetMaxRetries() => Options.MaxRetries;
```

Comment:
Change this to `CalculateRemainingRetries` accepting the current try count and returning Options.MaxRetries - currentTryCount.